### ISSUE TITLE
Implement cohort level Shifted BetaGeometric model

### ISSUE THREAD

**Issue by ricardoV94** [pymc-labs/pymc-marketing#169]
#133 implements the same model, at the user level granularity. 

It may make sense to implement a cohort level model. The posterior individual variables for costumers that enrolled at the same time are identical anyway, so it doesn't make sense to sample each one separately if the data comes in large cohorts.

Some of the summary statistics in #167 may also only make sense to the cohort level model.

**Comment by juanitorduz** [comment]
Do you mean the results from the paper http://brucehardie.com/notes/017/sBG_estimation.pdf ?

**Comment by ricardoV94** [comment]
I was not thinking about that specifically, just the original paper (one cohort over time). 

In the multiple cohort paper you mentioned they are just doing complete pooling across multiple cohorts right? Would be nice if we didn't need to do anything extra to allow that. Something like:

```python
alpha = pm.HalfNormal.dist(alpha)
beta = pm.HalfNormal.dist(beta)
cohort1 = clv.ShiftedBetaGeoCohortModel(..., alpha_prior=alpha, beta_prior=beta)
cohort2 = clv.ShiftedBetaGeoCohortModel(..., alpha_prior=alpha, beta_prior=beta)
cohorts = clv.combined_models(cohort1, cohort2)
```

And it would create a PyMC model that includes cohort1 and cohort2 as submodels, sharing the alpha and beta.
Calling `cohorts.fit()` would then provide the relevant part of the InferenceData to `cohort1` and `cohort2` so that you could use their special summary/plotting methods.

This would be neat because we could ultimately use more structured hyperpriors across models, even of different nature (e.g., correlation between lifetime and value).

Anyway... I was just talking about the single cohort model here

**Comment by juanitorduz** [comment]
Interesting! Makes sense! In addition, we could provide a hierarchical model?

**Comment by ColtAllen** [comment]
> [@ColtAllen](https://github.com/ColtAllen) I think we want to do a release this week so I would move the milestone to 0.17.0, ok?

Yeah post 0.16.0 is definitely more realistic haha. I'm testing the recursive `logp` of this model next week. After that checks out everything else is pretty much boilerplate and a tutorial notebook.
